File: client/src/app/dataModels/orders/makeOrder/makeOrderModel.ts
import { DeliveryType } from "../../enums/deliveryType";
import { PaymentMethod } from "../../enums/paymentMethod";
import { OrderCustomerInfo } from "./orderCustomerInfo";

export interface MakeOrderModel {
    cartItemsIds: string[];
    totalAmount: number;
    customerInfo: OrderCustomerInfo;
    deliveryType?: DeliveryType;
    paymentMethod?: PaymentMethod;
    customerNotes?: string;
    promocode?: string;
}

File: server/src/ApplicationCore/Models/Orders/MakeOrderModel.cs
using ApplicationCore.Enums;
using ApplicationCore.Models.Cart;

namespace ApplicationCore.Models.Orders;

public class MakeOrderModel
{
    public decimal TotalAmount { get; set; }
    
    public ICollection<Guid> CartItemsIds { get; set; }

    public OrderCustomerInfo CustomerInfo { get; set; }
    
    public DeliveryType DeliveryType { get; set; }
    
    public PaymentMethod PaymentMethod { get; set; }
    
    public string? CustomerNotes { get; set; }
    
    public string? PromoCode { get; set; }

    public decimal? PromoRate { get; set; }
    
    public decimal? PromoAmount { get; set; }
}